src/utils/cognito.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  AdminConfirmSignUpCommand,
+  AttributeType,
+  AuthenticationResultType,
+  CognitoIdentityProviderClientConfig,
+  InitiateAuthCommand,
+  SignUpCommand,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { getCognitoClient } from './internal';
+
+/**
+ * Get credentials for an existing user.
+ */
+export const cognitoSignIn = async (params: {
+  clientId: string;
+  username: string;
+  password: string;
+  config?: CognitoIdentityProviderClientConfig;
+}): Promise<AuthenticationResultType> => {
+  const { clientId, username, password, config } = params;
+  const client = getCognitoClient(config);
+  const { AuthenticationResult } = await client.send(
+    new InitiateAuthCommand({
+      AuthFlow: 'USER_PASSWORD_AUTH',
```

Comment:
This will not always work.
By default, the  `USER_PASSWORD_AUTH` flow is disabled I think. 
We need to support all flows, or at least `USER_SRP_AUTH`

I used this library in the past
https://www.npmjs.com/package/amazon-cognito-identity-js

it handles all that.

But the problem is that there might be challenges to answer, which is bad for automated processes.

So we might need to use the `AdminInitiateAuthCommand` instead.

Reply:
Yes, `USER_PASSWORD_AUTH` is turned off by default, so the current implementation will only work for Cognito client apps that support it But it's easy (and recommended in my opinion) to create a dedicated client app for "tests" that has `USER_PASSWORD_AUTH` (or `ADMIN_USER_PASSWORD_AUTH`) enabled, easy to setup and can be used right away.

Usually, tests will not run on production environments, so creating additional client apps will not cause any issues I believe.

I thought about adding other flows on demand (when requested by users), but this needs more work and effort the `USER_SRP_AUTH` flow, for example, requires a library that uses Secure Remote Password protocol like the [one](https://www.npmjs.com/package/amazon-cognito-identity-js) you sent.

> But the problem is that there might be challenges to answer, which is bad for automated processes.

+1 Yep, this is exactly the reason why I avoided other advanced & more secure flows and use the user/password flow.

Reply:
That is a valid argument. It probably makes sense to recommend creating a dedicated client for tests that use simple flows.
However, I think the MFA config, etc is not in the client, but "global" to the user pool.
Ther is probably no easy workaround to that, apart maybe admin auth flow.

I need to investigate more cognito flows, this is one of my weak points :)

Reply:
> So we might need to use the `AdminInitiateAuthCommand` instead.

Done switched to use `AdminInitiateAuthCommand`, https://github.com/serverless-guru/sls-jest/pull/73/commits/0aec2ce8d3bd9c42e733d06965353769c2f75e0f